src/IncludedCommands/Resources/PersistentBans.luau

```diff
@@ -149,20 +152,128 @@ end
+    return self:BanIds({
+        UserIds = {UserId},
+        Duration = -1,
+        DisplayReason = type(Message) == "string" and Message or "No ban reason specified.",
+        PrivateReason = type(Message) == "string" and Message or "No ban reason specified.",
+        ExcludeAltAccounts = false,
+        ApplyToUniverse = true
+    }, ModeratorId or 0, ModeratorName or "NexusAdmin")
+end
+
+--[[
+Bans user ids.
 --]]
-function PersistentBans.BanId(self: PersistentBans, UserId: number, Message: string?): ()
-    if not self.BansDataStore then return end
-    self.BansDataStore:Set(tostring(UserId), Message or true)
+function PersistentBans.BanIds(self: PersistentBans, Config: BanConfigType, ModeratorId: number?, ModeratorName: string?): (boolean, string, {number}, {number})
+    local CanBan, Immune = self:FilterIds(Config.UserIds, ModeratorId)
+    Config.UserIds = CanBan
+    if #Config.UserIds == 0 then
+        return false, "No affected UserIds", CanBan, Immune
+    end
+    self:Log(Config, ModeratorId, ModeratorName)
+    local Success, ErrorMessage = pcall(function()
+        Players:BanAsync(Config)
+    end)
+
+    return Success, ErrorMessage or "UserIds banned.", CanBan, Immune
 end
 
 --[[
 Unbans a user id.
+
+Preserved for backwards compatibility and minimal script changes.
 --]]
-function PersistentBans.UnbanId(self: PersistentBans, UserId)
-    if not self.BansDataStore then return end
-    self.BansDataStore:Set(tostring(UserId), nil)
+function PersistentBans.UnbanId(self: PersistentBans, UserId: number, ModeratorId: number?, ModeratorName: string?)
+    return self:UnbanIds({
+        UserIds = {UserId},
+        ApplyToUniverse = true,
+    }, ModeratorId or 0, ModeratorName or "NexusAdmin")
+end
+
+
+--[[
+Unbans user ids
+--]]
+function PersistentBans.UnbanIds(self: PersistentBans, Config: UnbanConfigType, ModeratorId: number?, ModeratorName: string?): (boolean, string?)
+    self:Log(Config, ModeratorId, ModeratorName)
+    return pcall(function()
+        Players:UnbanAsync(Config)
+    end)
 end
 
+--[[
+Logs an action
+--]]
+function PersistentBans.Log(self: PersistentBans, Config: BanConfigType | UnbanConfigType, ModeratorId: number?, ModeratorName : string?)
+    if self.LogsDataStore then
+        task.spawn(function()
+            if (Config :: BanConfigType).DisplayReason then
+                self.LogsDataStore:Update("PersistentBanActionsLog", function(OldLogs)
+                    OldLogs = OldLogs or {}
+                    table.insert(OldLogs, {
+                        Time = os.time(),
+                        Id = ModeratorId or 0,
+                        Name = ModeratorName or "NexusAdmin",
+                        Action = 0,
+                        EffectedUserIds = Config.UserIds,
+                        Duration = math.max(-1, Config.Duration),
+                        ExperienceWide = Config.ApplyToUniverse,
+                        IncludeAlts = not Config.ExcludeAltAccounts
```

Comment:
The typing is complaining that `Duration` and `ExcludeAltAccounts` are not part of `UnbanConfigType`.